translators/tests/test_crate.py

```diff
@@ -120,14 +131,69 @@ def test_benchmark_extended(translator):
     benchmark(translator, num_types=2, num_ids_per_type=2, num_updates=10, use_geo=True, use_time=True)
 
 
+def test_unsupported_ngsi_type(translator):
+    e = {
+        "type": "SoMeWeIrDtYpE",
+        "id": "sOmEwEiRdId",
+        TIME_INDEX_NAME: datetime(1970, 6, 28, 1, 1, 1).isoformat()[:-3],
+        "foo": {
+            "type": "DefinitivelyNotAValidNGSIType",
+            "value": "BaR",
+        }
+    }
+    # Assert exception is raised with error message containing both unsupported given type and supported expected type.
+    with pytest.raises(UnsupportedNGSIType, match=r"DefinitivelyNotAValidNGSIType.* Text"):
+        translator.insert([e])
+
+
+def test_capitals(translator):
+    entity_type = "SoMeWeIrDtYpE"
+    e = {
+        "type": entity_type,
+        "id": "sOmEwEiRdId",
+        TIME_INDEX_NAME: datetime.now().isoformat()[:-3],
+        "Foo": {
+            "type": "Text",
+            "value": "FoO",
+        },
+        "bAr": {
+            "type": "Text",
+            "value": "bAr",
+        },
+    }
+    translator.insert([e])
+    translator._refresh([entity_type])
+    entities = translator.query()
+    assert len(entities) == 1
+    assert_ngsi_entity_equals(e, entities[0])
+
+    # If a new attribute comes later, I want it translated as well.
+    e2 = e.copy()
+    e2['id'] = 'SOmEwEiRdId2'
+    e2['NewAttr'] = {"type": "Text", "value": "NewAttrValue!"}
+    e2[TIME_INDEX_NAME] = datetime.now().isoformat()[:-3]
+
+    translator.insert([e2])
+    translator._refresh([entity_type])
+    entities = translator.query()
+    assert len(entities) == 2
+
+    assert_ngsi_entity_equals(e2, entities[1])
+    # Note that old entity gets None for the new attribute
+    e['NewAttr'] = {'type': 'Text', 'value': None}
+    assert_ngsi_entity_equals(e, entities[0])
+
+
 # FIWARE DATA MODELS
 
```

Comment:
we need to introduce more tests, at least all the ones that now have a decent schema definition.

please include:
* https://github.com/Fiware/dataModels/tree/master/Alert
* https://github.com/Fiware/dataModels/tree/master/WasteManagement
* https://github.com/Fiware/dataModels/tree/master/Device
* https://github.com/Fiware/dataModels/tree/master/PointOfInterest
* https://github.com/Fiware/dataModels/tree/master/Transportation/Bike/BikeHireDockingStation

Reply:
Created https://github.com/smartsdk/ngsi-timeseries-api/issues/33 to be done in a new PR